Liquidity-provider-server automated update of documentation update-from-lps-20260224-223005 (#473)
rsksmart/devportal: rif/identity/data-vault/cpinner/cpinner-client.markdown

---
layout: rsk
---

## Data vault - Centralized pinner client

A Web Client to simplify the way the services provided by the IPFS Centralized Data Vault Service are consumed.

### Features

- Manage authentication according to the [DID Auth protocol](https://rsksmart.github.io/rif-identity-docs/ssi/libraries/express-did-auth)

> - CRUD operations over the RIF Data Vault
> 
> - Stores the authentication credentials in the given storage 

Reviewer: ```suggestion
- CRUD operations over the RIF Data Vault
- Stores the authentication credentials in the given storage 
```

### Usage

#### Instantiate

The package expose a `DataVaultWebClient` class that receives a `Config` object when it is instantiated.

The `Config` object has the following fields:

> `serviceUrl: string`: the IPFS Centralized Data Vault Service url
> `serviceDid?: string`: the IPFS Centralized Data Vault Service url did. It is required if the package will be used to perform authenticated requests (Create, Swap or Delete). This field will be used to compare with the issuer of the access token.
> `did?: string`: the client did. It is required if performing authenticated requests
> `rpcPersonalSign?: (data: string) => string`: the rpcPersonalSign function associated to the client did. It is used to sign the challenge in the login process. Metamask example: `(data: string) => window.ethereum.request({ method: 'personal_sign', params: [address, data] })`
> `storage?: ClientKeyValueStorage`: object that MUST implement the `ClientKeyValueStorage` interface. It is used to save the `accessToken` and `refreshToken`. It uses [window.localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) if not storage object provided.

Reviewer: ```suggestion
- `serviceUrl: string`: the IPFS Centralized Data Vault Service url
- `serviceDid?: string`: the IPFS Centralized Data Vault Service url did. It is required if the package will be used to perform authenticated requests (Create, Swap or Delete). This field will be used to compare with the issuer of the access token.
- `did?: string`: the client did. It is required if performing authenticated requests
- `rpcPersonalSign?: (data: string) => string`: the rpcPersonalSign function associated to the client did. It is used to sign the challenge in the login process. Metamask example: `(data: string) => window.ethereum.request({ method: 'personal_sign', params: [address, data] })`
- `storage?: ClientKeyValueStorage`: object that MUST implement the `ClientKeyValueStorage` interface. It is used to save the `accessToken` and `refreshToken`. It uses [window.localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) if not storage object provided.
```

```

#### Delete

It deletes the `content` associated to the given `key` (and `id` if provided).
It receives an object with `key` as the mandatory field. It also allows to add the `id` of the desired `content` to be deleted.
If `id` is not provided, it will delete ALL the `content` associated to the `key`.
If `id` is provided, it just deletes the `content` that matches the given `id` and `key`

If the `key` does not exist, will create it.

If the client is not logged in yet, it will log it in prior to deleting, so it may prompt the user to sign the login message

##### Without `id`

```typescript
import DataVaultWebClient from '@rsksmart/ipfs-cpinner-client'

const client = new DataVaultWebClient({ serviceUrl, did, signer, serviceDid })

const key = 'MyKey'

await client.delete({ key })
```

##### With `id`

```typescript
import DataVaultWebClient from '@rsksmart/ipfs-cpinner-client'

const client = new DataVaultWebClient({ serviceUrl, did, signer, serviceDid })

const key = 'MyKey'
const content = 'this is my content'

const id = await client.create({ key, content: original1 })

await client.delete({ key, id })
```

### Run for development

> See our repo in [Github](https://github.com/rsksmart/rif-data-vault/tree/master/modules/ipfs-cpinner-client).

Reviewer: ```suggestion
Check out [`ipfs-cpinner-client` in the `rif-data-vault` repo](https://github.com/rsksmart/rif-data-vault/tree/master/modules/ipfs-cpinner-client).
```
